src/sisl/io/vasp/tests/test_chg.py

```diff
@@ -40,3 +40,53 @@ def test_graphene_chgcar_index_float(sisl_files):
     gridh = chgSileVASP(f).read_grid(index=[0.5])
 
     assert grid.grid.sum() / 2 == pytest.approx(gridh.grid.sum())
+
+
+def test_nitric_oxide_chg(sisl_files):
```

Comment:
here you would be better of using fixtures to re-use the semantics.
```python
@pytest.fixture(scope="module", params=["unpol", "pol", "soi"])
def spin_conf(request):
    return request.param

def test_nitric_oxide_chg(sisl_files, spin_conf):
    #spin_conf will now be one of the above elements
```

untested, but should work